go/oasis-test-runner/scenario/e2e/e2e.go

```diff
@@ -12,52 +12,64 @@ import (
 )
 
 const (
+	// cfgNodeBinary is the path to oasis-node executable.
 	cfgNodeBinary = "node.binary"
 )
 
 var (
-	// E2eParamsDummy is a dummy instance of e2eImpl used to register e2e-wise parameters.
-	E2eParamsDummy *e2eImpl = &e2eImpl{name: "e2e"}
-
-	logger = logging.GetLogger("e2e/common")
+	// E2eParamsDummy is a dummy instance of e2eImpl used to register global e2e flags.
+	E2eParamsDummy *e2eImpl = newE2eImpl("")
 )
 
 // e2eImpl is a base class for tests involving oasis-node.
 type e2eImpl struct {
 	net    *oasis.Network
 	name   string
 	logger *logging.Logger
-
-	// nodeBinary is the path to oasis-node executable.
-	nodeBinary string
+	flags  *flag.FlagSet
 }
 
 func newE2eImpl(name string) *e2eImpl {
-	return &e2eImpl{
-		name:       "e2e/" + name,
-		logger:     logging.GetLogger("scenario/e2e/" + name),
-		nodeBinary: "oasis-node",
+	// Empty scenario name is used for registering global parameters only.
+	fullName := "e2e"
+	if name != "" {
+		fullName += "/" + name
+	}
+
+	sc := &e2eImpl{
+		name:   fullName,
+		logger: logging.GetLogger("scenario/" + fullName),
+		flags:  newE2eFlags(fullName),
 	}
+
+	return sc
+}
+
+func newE2eFlags(name string) *flag.FlagSet {
+	fs := flag.NewFlagSet(name, flag.ContinueOnError)
+	fs.String(cfgNodeBinary, "oasis-node", "path to the node binary")
+
+	return fs
 }
 
 func (sc *e2eImpl) Clone() e2eImpl {
-	return e2eImpl{
-		net:        sc.net,
-		name:       sc.name,
-		logger:     sc.logger,
-		nodeBinary: sc.nodeBinary,
+	newSc := &e2eImpl{
+		net:    sc.net,
+		name:   sc.name,
+		logger: sc.logger,
+		flags:  newE2eFlags(sc.name),
 	}
+	scenario.CopyFlagValues(newSc.flags, sc.flags)
```

Comment:
This seems slightly awkward, can you instead define your own flagset like this:
```golang
type FlagSet struct {
    flag.FlagSet
}
```
And then define a `Clone()` method on it that returns a cloned flagset? Put it under the `env` package.

Actually I see that you already do this for `env.ParameterFlagSet`, why not use that type everywhere (e.g., make `Parameters()` return a `*ParameterFlagSet`) and add a `Clone()` method?

Reply:
I pushed a draft version of the Clone() method. This seems to work, but it's ugly - I need to use `reflect` package to copy the interface.

Reply:
cleaned up, ready for review